lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/IMRUCloseTaskTest.cs

```diff
@@ -30,24 +34,22 @@ namespace Org.Apache.REEF.Tests.Functional.IMRU
 {
     /// <summary>
     /// This is to test close event handler in IMRU tasks
-    /// The test provide IRunningTask, IFailedTask and ICOmpletedTask handlers so that to trigger close events and handle the 
+    /// The test provide IRunningTask, IFailedTask and ICompletedTask handlers so that to trigger close events and handle the 
     /// failed tasks and completed tasks
     /// </summary>
     [Collection("FunctionalTests")]
     public class IMRUCloseTaskTest : IMRUBrodcastReduceTestBase
     {
         private const string CompletedTaskMessage = "CompletedTaskMessage";
+        private const string FailEvaluatorMessage = "FailEvaluatorMessage";
         private const string FailTaskMessage = "FailTaskMessage";
 
         /// <summary>
         /// This test is for running in local runtime
         /// It sends close event for all the running tasks.
-        /// It first informs the Call method to stop.
-        /// If Call method is running properly, it will respect to this flag and will return properly, that will end up ICompletedTask event.
-        ////If Call method is hung some where and cannot be returned, the close handler will throw exception, that would cause IFailedTask event.
-        /// As we are testing IMRU Task not a test task, the behavior is not deterministic. It can be CompletedTask or FailedTask
-        /// No matter how the task is closed, the total number of completed task and failed task should be equal to the 
-        /// total number of the tasks.
+        /// In the task close handler, the cancellation token will be set, resulting the task
+        /// returned from the Call() method. 
+        /// Expect all the 4 tasks are completed. No Failed task and no failed Evaluator. 
```

Comment:
In all the exceptions cases in the IMRU tasks, as long as the cancellation token is set, the task will be returned from Call() properly. 
For the cases where IOException or TcpClientConnectionException are thrown, if cancellation token is not set yet, exception will be thrown and Task will fail. I do see IOException and TcpClientConnectionException in testing , but haven't seen the case where cancellation token is not set yet. It might happen in race condition. 
I can add the failed task back just in case.